modules/client/src/client-common/modules/web3.ts

```diff
@@ -154,51 +90,34 @@ export class Web3Module implements IClientWeb3Core {
    */
   public attachContract<T>(
     address: string,
-    abi: ContractInterface
+    abi: ContractInterface,
   ): Contract & T {
-    if (!address) throw new Error("Invalid contract address");
-    else if (!abi) throw new Error("Invalid contract ABI");
-
-    const signer = this.getSigner();
-    if (!signer && !this.getProvider()) {
-      throw new Error("No signer");
-    }
-
-    const provider = this.getProvider();
-    if (!provider) throw new Error("No provider");
-
-    const contract = new Contract(address, abi, provider);
-
-    if (!signer) {
-      return contract as Contract & T;
-    } else if (signer instanceof Wallet) {
-      return contract.connect(signer.connect(provider)) as Contract & T;
-    }
-
-    return contract.connect(signer) as Contract & T;
+    if (!address || !isAddress(address)) throw new InvalidAddressError();
+    else if (!abi) throw new InvalidContractAbiError();
+    const signer = this.getConnectedSigner();
+    return new Contract(address, abi, signer) as Contract & T;
```

Comment:
Are we sure that this line of code covers all the `connect` logic that existed before? That is a bit tricky from ethers, everything has to be sort of double connected for things to work properly

Reply:
- The address is checked
- The abi is checked
- The getConnectedSigner calls this.getSigner that thows if its null, so the signer is checked
- The getConnectedSigner also calls this.getProvider that throws if its null, so the provider is checked too
- Then we checked connected the signer with the provider, using the `getConnectedSigner` returns a signer connected with the provider
- We created the contract with the `provider` and then we connected to the signer, here we provide a connected signer so we can skip that. 

I think we are good